store/src/java/com/zimbra/cs/dav/DavContext.java

```diff
@@ -113,6 +113,7 @@ public static void removeExtension(Extension ex) {
     private String mCollectionPath;
     private RequestProp mResponseProp;
     private String mDavCompliance;
+    public static boolean noFileExtension;
```

Comment:
@dasiyogesh this boolean flag we have specifically added to check at FileUpload class if the file has extension or not
FileUploadServlet is used extensively so adding a new flag to it would have been difficult.
This flag is false by default, it is turned true only in this class and once processed at fileUploadServlet it is turned back to false to make sure there is no impact on existing flow.